igvc_navigation/src/navigation_server/navigation_server.cpp

```diff
@@ -147,35 +151,115 @@ void NavigationServer::actionGetPathDone(const actionlib::SimpleClientGoalState
         navigate_waypoint_result.outcome = result.outcome;
         navigate_waypoint_result.message = result.message;
         ROS_WARN_STREAM_NAMED("nav_server", "Abort the execution of the planner: " << result.message);
-        current_goal_handle_.setAborted(navigate_waypoint_result, state.getText());
-        current_state_ = FAILED;
+        goal_handle_.setAborted(navigate_waypoint_result, state.getText());
+        navigation_state_ = FAILED;
       }
       break;
     case actionlib::SimpleClientGoalState::PREEMPTED:
       // the get_path action has been preempted.
       // copy result from get_path action
       navigate_waypoint_result.outcome = result.outcome;
       navigate_waypoint_result.message = result.message;
-      current_goal_handle_.setCanceled(navigate_waypoint_result, state.getText());
-      current_state_ = CANCELED;
+      goal_handle_.setCanceled(navigate_waypoint_result, state.getText());
+      navigation_state_ = CANCELED;
       break;
     case actionlib::SimpleClientGoalState::LOST:
-      ROS_FATAL_STREAM_NAMED("nav_server", "Connection lost to the action \"get_path\"!");
-      current_goal_handle_.setAborted();
-      current_state_ = FAILED;
+      ROS_FATAL_STREAM_NAMED("nav_server", "Connection lost to the action 'get_path'!");
+      goal_handle_.setAborted();
+      navigation_state_ = FAILED;
       break;
     default:
       ROS_FATAL_STREAM_NAMED("nav_server", "Unreachable state reached for get_path: " << state.toString());
-      current_state_ = FAILED;
+      navigation_state_ = FAILED;
       break;
   }
+
+  // only start replanning if we are executing path
+  if (navigation_state_ == EXE_PATH)
+  {
+    current_replanning_tries_ = 0;
+    // prevent actionGetPathReplanningDone from replanning until new goal is set
+    std::lock_guard<std::mutex> guard(replanning_mtx_);
+    // reset replanning rate and sleep for remaining time
+    replanning_rate_.reset();
+    replanning_rate_.sleep();
+    // check to see if still executing path and that get_path is done
+    if (navigation_state_ == EXE_PATH &&
+        action_client_get_path_.getState() != actionlib::SimpleClientGoalState::PENDING &&
+        action_client_get_path_.getState() != actionlib::SimpleClientGoalState::ACTIVE)
+    {
+      ROS_INFO_STREAM_NAMED("nav_server", "Start replanning, using the 'get_path' action!");
+      action_client_get_path_.sendGoal(get_path_goal_,
+                                       boost::bind(&NavigationServer::actionGetPathReplanningDone, this, _1, _2));
+    }
+  }
+}
+
+void NavigationServer::actionGetPathReplanningDone(const actionlib::SimpleClientGoalState &state,
+                                                   const mbf_msgs::GetPathResultConstPtr &result_ptr)
+{
+  // check if we are executing path
+  if (navigation_state_ == EXE_PATH)
+  {
+    const mbf_msgs::GetPathResult &result = *(result_ptr.get());
```

Comment:
`GetPathResultConstPtr` is a `boost::shared_ptr`, so you should be able to just dereference it without doing `*(result_ptr.get())`